benchmarks/src/main/java/io/grpc/benchmarks/qps/Utils.java

```diff
@@ -129,10 +131,27 @@ static SimpleRequest newRequest(ClientConfiguration config) {
   static Channel newClientChannel(ClientConfiguration config) throws IOException {
     if (config.transport == ClientConfiguration.Transport.OK_HTTP) {
       InetSocketAddress addr = (InetSocketAddress) config.address;
-      return OkHttpChannelBuilder
+      OkHttpChannelBuilder builder = OkHttpChannelBuilder
           .forAddress(addr.getHostName(), addr.getPort())
-          .executor(config.directExecutor ? MoreExecutors.newDirectExecutorService() : null)
-          .build();
+          .executor(config.directExecutor ? MoreExecutors.newDirectExecutorService() : null);
+      if (config.tls) {
+        SSLSocketFactory factory;
+        if (config.testca) {
+          builder.overrideHostForAuthority(TestUtils.TEST_SERVER_HOST);
```

Comment:
You shouldn't hard-code the override. Instead, pass `addr.getAddress().getHostAddress()` to `forAddress()` and always call `overrideHostForAuthority()` with `addr.getHostString()`. Yes, getHostString() is only in Java 7, but it doesn't seem bad to depend on Java 7 here.